src/UrlHelper.php

```diff
@@ -10,12 +10,21 @@
 use Psr\Http\Message\ServerRequestInterface;
 
 use function array_merge;
+use function assert;
 use function count;
 use function http_build_query;
+use function is_string;
 use function ltrim;
 use function preg_match;
 use function sprintf;
 
+/**
+ * @psalm-type UrlGeneratorOptions = array{
+ *     router?: array<array-key, mixed>,
+ *     reuse_result_params?: bool,
+ *     reuse_query_params?: bool,
```

Comment:
Should we declare `...` here?

Reply:
I don't think we should - it's a small and well-defined set of options and it forces maintainers to document new options if and when they are added.